crowdsourcing/serializers/payment.py

```diff
@@ -0,0 +1,177 @@
+    amount = serializers.IntegerField()
+    type = serializers.ChoiceField(choices=['self', 'other'])
+    username = serializers.SlugRelatedField(queryset=User.objects.all(), slug_field='username', allow_null=True)
+    method = serializers.ChoiceField(choices=['paypal', 'credit_card'])
+    credit_card = CreditCardSerializer(many=False, required=False)
+
+    class Meta:
+        validators = [
+            InequalityValidator(
+                field='amount', operator='gt', value=1
+            ),
+            ConditionallyRequiredValidator(field='method', field2='credit_card', value='credit_card')
+        ]
+
+    def build_payment(self, *args, **kwargs):
+        host = ''
+        if self.context['request'].is_secure():
+            host = 'https://'
+        else:
+            host = 'http://'
+        host += self.context['request'].META['HTTP_HOST']
+
+        payment = {"intent": "sale",
+                   "payer": {
+                       "payment_method": self.validated_data['method']
+                   },
+                   "redirect_urls": {
+                       "return_url": host+"/paypal-success",
+                       "cancel_url": host+"/paypal-cancelled"
+                   },
+                   "transactions": [{
+                       "item_list": {
+                           "items": [{
+                               "name": "Daemo Deposit.",
+                               "sku": "DMO-7CA000",
+                               "price": self.validated_data['amount'],
+                               "currency": "USD",
+                               "quantity": 1}]},
+                       "amount": {
+                           "total": self.validated_data['amount'],
+                           "currency": "USD"},
+                       "description": "Daemo Deposit."}]
+                   }
+
+        if self.validated_data['method'] == 'credit_card':
+            payment['payer']['funding_instruments'] = [{
+                "credit_card": self.validated_data['credit_card']
+            }]
+
+        return payment
+
+    def create(self, *args, **kwargs):
+        recipient = None
+        recipient_profile = None
+        payment_data = self.build_payment()
+        if self.validated_data['type'] == 'self':
+            recipient_profile = self.context['request'].user.userprofile
+        else:
+            recipient_profile = get_model_or_none(UserProfile, user__username=self.validated_data['username'])
+        recipient = get_model_or_none(FinancialAccount, owner=recipient_profile, type='requester')
+        paypalbackend = PayPalBackend()
+        payment = paypalbackend.paypalrestsdk.Payment(payment_data)
+        if payment.create():
+            redirect_url = next((link for link in payment['links'] if link['method'] == 'REDIRECT'), '#')
+            flow_data = {
+                "redirect_url": redirect_url['href'],
+                "paypal_id": payment.id
+            }
```

Comment:
I think you're crashing here when the method is a credit card because I believe you aren't getting back a redirect_url and so redirect_url['href'] breaks.